### PR TITLE
Assets ignore lists

### PR DESCRIPTION
**Pull request by brunobar79**
To prevent assets to be "re-detected" after being removed, we need to keep them in state.
I've implemented that by adding two additional arrays to the AssetDetection controller to keep track of removed collectibles and assets.

Also added to additional methods to manually clear those lists if required

### CODE REVIEW

File: src/AssetsController.ts [hunk 1]
```diff
@@ -419,13 +423,26 @@ export class AssetsController extends BaseController<AssetsConfig, AssetsState>
 		address = toChecksumAddress(address);
 		const { allCollectibles, collectibles } = this.state;
 		const { networkType, selectedAddress } = this.config;
-		const newCollectibles = collectibles.filter(
-			(collectible) => !(collectible.address === address && collectible.tokenId === tokenId)
+		const ignoredCollectible = collectibles.find(
+			(collectible) => collectible.address === address && collectible.tokenId === tokenId
 		);
+
+		const ignoredCollectibles = [...this.state.ignoredCollectibles];
```
Comment (estebanmino) [inline src/AssetsController.ts]: curious why did you need to do this?

File: src/AssetsController.ts [hunk 2]
```diff
@@ -419,13 +423,26 @@ export class AssetsController extends BaseController<AssetsConfig, AssetsState>
...
+		const newCollectibles = collectibles.filter((collectible) => {
+			return !(collectible.address === address && collectible.tokenId === tokenId);
+		});
+
 		const addressCollectibles = allCollectibles[selectedAddress];
 		const newAddressCollectibles = { ...addressCollectibles, ...{ [networkType]: newCollectibles } };
 		const newAllCollectibles = { ...allCollectibles, ...{ [selectedAddress]: newAddressCollectibles } };
-		this.update({ allCollectibles: newAllCollectibles, collectibles: newCollectibles });
+
+		this.update({ ignoredCollectibles });
```
Comment (estebanmino) [inline src/AssetsController.ts]: didn't you have issues from the app doing this? we used to have a bug with states being updated like this before where states weren't being updated. We solved it changing all `update`s to the format `this.update({ ignoredCollectibles: [...ignoredCollectibles]})`

Comment (estebanmino) [reply src/AssetsController.ts]: also we could add it to the `update` below to avoid two state updates

File: src/AssetsDetectionController.ts [hunk 3]
```diff
@@ -195,16 +204,28 @@ export class AssetsDetectionController extends BaseController<AssetsDetectionCon
...
-					description,
-					image: image_preview_url,
-					name
-				},
-				true
-			);
+
+			let ignored = null;
+			/* istanbul ignore else */
+			if (ignoredCollectibles && ignoredCollectibles.length) {
```
Comment (estebanmino) [inline src/AssetsDetectionController.ts]: after our discussion `ignoredCollectibles` should be always at least an empty array

File: src/AssetsController.ts [hunk 4]
```diff
@@ -651,13 +672,20 @@ export class AssetsController extends BaseController<AssetsConfig, AssetsState>
...
 		address = toChecksumAddress(address);
-		const { allTokens, tokens } = this.state;
+		const { allTokens, tokens, ignoredTokens } = this.state;
 		const { networkType, selectedAddress } = this.config;
-		const newTokens = tokens.filter((token) => token.address !== address);
+		const newIgnoredTokens = [...ignoredTokens];
+		const newTokens = tokens.filter((token) => {
+			/* istanbul ignore else */
+			if (token.address === address) {
+				newIgnoredTokens.push(token);
```
Comment (estebanmino) [inline src/AssetsController.ts]: I believe here (and doing the same for collectibles) would duplicate `ignoredTokens` if he user decides to remove a token twice or more (edge case), there is no validation that a token is already on that list.

Comment (brunobar79) [reply src/AssetsController.ts]: Added check to prevent dupes and added tests for that scenario
